nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -119,7 +119,14 @@ let
         "systemd-modules-load.service"
         "systemd-ask-password-console.service"
       ];
-      wantedBy = (getPoolMounts prefix pool) ++ [ "local-fs.target" ];
+      requiredBy = if neededForBoot then
+        (getPoolMounts prefix pool) ++ [ "local-fs.target" ]
+      else
+        [ ];
+      wantedBy = if !neededForBoot then
+        (getPoolMounts prefix pool) ++ [ "local-fs.target" ]
+      else
+        [ ];
```

Comment:
See #227208

In that PR, it is changed so that the import service is *always* required by the relevant mount units, and required by `zfs-import.target` instead of `local-fs.target`. Whether or not this will lead to a failed boot when the pool fails to import is then dependent on whether or not the mount is required for boot (i.e. doesn't have the `nofail` option). So I think that PR covers this in a more correct way.

Reply:
Ah cool. Didn't consider the ordering before `local-fs.target` to be incorrect in the first place.
The service activation ordering was still a bit weird (but it worked) with my approach, so that explains it.

I removed the service related changes from this PR and have it focus solely on the unlock timeout.